app/routes.py

```diff
@@ -1,2 +1,93 @@
-from flask import Blueprint
+import re
+from flask import Blueprint, jsonify, abort, make_response, request
+from app import db
+from app.models.Planet import Planet
 
+planet_bp = Blueprint("planet", __name__, url_prefix="/planets")
+
+#Validate planet data
+def validate_planet(planet_id):
+    try:
+        planet_id = int(planet_id)
+    except:
+        return abort(make_response({"message": f"Planet {planet_id} is invalid"}, 400))
+
+    planet = Planet.query.get(planet_id)
+    
+    if not planet:       
+        return abort(make_response({"message": f"Planet {planet_id} is not found"}, 404))
+    return planet
+
+#GET all planets or GET by NAME query
+@planet_bp.route("", methods=["GET"])
+def get_all_planets():
+    name_query = request.args.get("name").title()
+    if name_query:
+        planets = Planet.query.filter_by(name=name_query)
+    else:
+        planets = Planet.query.all()
+
+    planet_response = []
+    for planet in planets:
+        planet_response.append({
+            "id" : planet.planet_id,
+            "name" : planet.name,
+            "description" : planet.description,
+            "color" : planet.color
+        })
+    if not planet_response:
+        return abort(make_response({"message": f"Planet {name_query} is not found"}, 404))
+    
+    return jsonify(planet_response)
```

Comment:
don't forget your status code!

```suggestion
    return jsonify(planet_response), 200
```